src/plugins/vis_default_editor/public/agg_filters/agg_type_filters.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IAggType, IAggConfig, IndexPattern, search } from '../../../data/public';
+
+const { propFilter } = search.aggs;
+const filterByName = propFilter('name');
+
+type AggTypeFilter = (
+  aggType: IAggType,
+  indexPattern: IndexPattern,
+  aggConfig: IAggConfig,
+  aggFilter: string[]
+) => boolean;
+
+const filters: AggTypeFilter[] = [
+  /**
+   * This filter checks the defined aggFilter in the schemas of that visualization
+   * and limits available aggregations based on that.
+   */
+  (aggType, indexPattern, aggConfig, aggFilter) => {
+    const doesSchemaAllowAggType = filterByName([aggType], aggFilter).length !== 0;
+    return doesSchemaAllowAggType;
+  },
+  /**
+   * If rollup index pattern, check its capabilities
+   * and limit available aggregations based on that.
+   */
+  (aggType, indexPattern, aggConfig, aggFilter) => {
+    if (indexPattern.type !== 'rollup') {
+      return true;
+    }
+    const aggName = aggType.name;
+    const aggs = indexPattern.typeMeta && indexPattern.typeMeta.aggs;
+
+    // Return doc_count (which is collected by default for rollup date histogram, histogram, and terms)
+    // and the rest of the defined metrics from capabilities.
+    return aggName === 'count' || (!!aggs && Object.keys(aggs).includes(aggName)) || false;
+  },
```

Comment:
We are trying to avoid usage of `rollup` references in oss code, so there was `getAggregationRestrictions` method created in `IndexPattern` class to check if there are any restrictions in index pattern. You could take a look at example done in `src/plugins/vis_default_editor/public/components/utils/editor_config.ts`.
Finally, this could look like this:
```
  /**
   * Check index pattern aggregation restrictions
   * and limit available fields for a given aggType based on that.
   */
  (field, aggConfig) => {
    const indexPattern = aggConfig.getIndexPattern();
    const aggRestrictions = indexPattern.getAggregationRestrictions();

    if (!aggRestrictions) {
      return true;
    }

    const aggName = aggConfig.type && aggConfig.type.name;
    const aggFields = aggRestrictions[aggName];
    return !!aggFields && !!aggFields[field.name];
  },
```
The same could be done also for `agg_type_filters` file.

Reply:
Ah I totally forgot about `getAggregationRestrictions` -- thanks for the reminder! Will update accordingly.